### PR TITLE
feat: throw error based on login ✨

### PR DESCRIPTION
**Pull request by gregorylegarec**
The goal of this PR is to be able to test the different kind of konnector errors in Collect.

If the login match a [known konnector error](https://github.com/cozy/cozy-konnector-debug/pull/2/files#diff-168726dbe96b3ce427e7fedce31bb0bcR6), the corresponding error is thrown.

### CODE REVIEW

**Comment by gregorylegarec** [comment]
[(3) [COL] ETU j'ai différents messages en fonction des erreurs des connecteurs [2]](https://trello.com/c/DNBu5P4E/874-3-col-etu-jai-diff%C3%A9rents-messages-en-fonction-des-erreurs-des-connecteurs-2)

File: index.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+  'USER_ACTION_NEEDED.ACCOUNT_REMOVED',
+  'USER_ACTION_NEEDED.CHANGE_PASSWORD',
+  'VENDOR_DOWN.BANK_DOWN',
+  'VENDOR_DOWN.LINXO_DOWN'
+];
 
 module.exports = new BaseKonnector(fields => {
-  const timeout = Number(fields.timeout) || 8000
+  if (!!fields.login && konnectorErrors.includes(fields.login.toUpperCase())) {
+    throw new Error(fields.login.toUpperCase())
```
Comment (doubleface) [inline index.js]: I think it would be better to throw this error after the timeout

**Review by doubleface — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (gregorylegarec) [reply index.js]: Is there any specific reason ? Using Debug konnector frequently I am pretty bored by waiting every time :D

Comment (doubleface) [reply index.js]: Then you can fill the timeout field... But since you will be the main user of this connector, I am Ok with it.

**Review by doubleface — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (gregorylegarec) [reply index.js]: Ah ok I see, I did not see this field was here. Actually it is not defined in Collect's `konnectors.json`. But we can fix it.

Comment (doubleface) [reply index.js]: Ok, I did not check the konnectors.json, I understand.

Comment (gregorylegarec) [reply index.js]: I've just added a dedicated commit.

**Review by CPatchane — APPROVED** [review]
Great 👍

**Comment by CPatchane** [comment]
Thanks @gregorylegarec 🍂

File: index.js (hunk 1 above)
Comment (gregorylegarec) [reply index.js]: @doubleface https://github.com/cozy/cozy-collect/pull/603

**Pull request merged**